examples/cifar10_cnn_capsule.py

```diff
@@ -0,0 +1,214 @@
+"""
+Train a simple CNN-Capsule Network on the CIFAR10 small images dataset.
+
+Without Data Augmentation:
+It gets to 75% validation accuracy in 10 epochs,
+and 79% after 15 epochs, and overfitting after 20 epcohs
+
+With Data Augmentation:
+It gets to 75% validation accuracy in 10 epochs,
+and 79% after 15 epochs, and 83% after 30 epcohs.
+In my test, highest validation accuracy is 83.79% after 50 epcohs.
+
+This is a fast Implement, just 20s/epcoh with a gtx 1070 gpu.
+"""
+
+from __future__ import print_function
+from keras import backend as K
+from keras.engine.topology import Layer
+from keras.layers import Activation
+from keras import utils
+from keras.datasets import cifar10
+from keras.models import Model
+from keras.layers import *
+from keras.preprocessing.image import ImageDataGenerator
+
+
+# a squashing function. but it has litte difference from the Hinton's paper.
+# it seems that this form of squashing performs better.
+def squash(x, axis=-1):
+    s_squared_norm = K.sum(K.square(x), axis, keepdims=True) + K.epsilon()
+    scale = K.sqrt(s_squared_norm) / (0.5 + s_squared_norm)
+    return scale * x
+
+
+# define our own softmax function instead of K.softmax
+def softmax(x, axis=-1):
+    ex = K.exp(x - K.max(x, axis=axis, keepdims=True))
+    return ex / K.sum(ex, axis=axis, keepdims=True)
+
+
+# define the margin loss like hinge loss
+def margin_loss(y_true, y_pred):
+    lamb, margin = 0.5, 0.1
+    return y_true * K.square(K.relu(1 - margin - y_pred)) + lamb * (
+        1 - y_true) * K.square(K.relu(y_pred - margin))
+
+
+class Capsule(Layer):
+    """
+    A Capsule Implement with Pure Keras
+    There are two vesions of Capsule.
+    One is like dense layer (for the fixed-shape input),
+    and the other is like timedistributed dense (for various length input).
+    The input shape of Capsule must be (batch_size,
+                                        input_num_capsule,
+                                        input_dim_capsule
+                                       )
+    and the output shape is (batch_size,
+                             num_capsule,
+                             dim_capsule
+                            )
+
+    Capsule Implement is from https://github.com/bojone/Capsule/
+    Capsule Paper: https://arxiv.org/abs/1710.09829
+    """
+    def __init__(self,
+                 num_capsule,
+                 dim_capsule,
+                 routings=3,
+                 share_weights=True,
+                 activation='default',
+                 **kwargs):
+        super(Capsule, self).__init__(**kwargs)
+        self.num_capsule = num_capsule
+        self.dim_capsule = dim_capsule
+        self.routings = routings
+        self.share_weights = share_weights
+        if activation == 'default':
+            self.activation = squash
+        else:
+            self.activation = Activation(activation)
```

Comment:
This should be `activations.get(activation)` where `activations` is `from keras import activations`